### PR TITLE
Support for provider autocompletion

### PR DESCRIPTION
**Pull request by p-v**
The provider list is rapidly increasing. So, thought of adding autocompletion to it. Let me know your view on this.

### CODE REVIEW

**Comment by zquestz** [comment]
Autocomplete would be awesome.

**Comment by KeizerDev** [comment]
That's really cool! But isn't there a more 'go' way?

**Comment by zquestz** [comment]
This actually looks great to me. There isn't a good way to do this in go as the shell is responsible for the expansion. One thing I would like to see though is zsh support, as I no longer use bash. =)

File: README.md [hunk 1]
```diff
@@ -51,6 +51,23 @@ Or toasters on amazon.
...
+Autocompletion for providers is supported. For setting up autocompletion, 
+
+1. Have `s` installed
+2. Add the following lines to `~/.bash_profile`
+```
+if [ -f /path/to/s-completion.bash ]; then
+    . /path/to/s-completion.bash
+fi
+```
+And you are good to go.
```
Comment (zquestz) [inline README.md]: NIT: "Now you are good to go."

File: README.md [hunk 2]
```diff
@@ -51,6 +51,23 @@ Or toasters on amazon.
 s -p am toasters
 ```
 
+## Provider Autocompletion
+
+Autocompletion for providers is supported. For setting up autocompletion,
```
Comment (zquestz) [inline README.md]: I would do ":" instead of ","

**Comment by zquestz** [comment]
Some tiny nits, I can go ahead and change those. I think this is good to merge. We can worry about zsh another time. =)

**Comment by zquestz** [comment]
Feel free to add a PR if you want to be in the README. Appreciate the help!

File: README.md [hunk 3]
```diff
@@ -51,6 +51,23 @@ Or toasters on amazon.
...
 
+## Provider Autocompletion
+
+Autocompletion for providers is supported. For setting up autocompletion, 
+
+1. Have `s` installed
+2. Add the following lines to `~/.bash_profile`
+```
+if [ -f /path/to/s-completion.bash ]; then
+    . /path/to/s-completion.bash
```
Comment (KeizerDev) [inline README.md]: Why not 
` . $GOPATH/src/github.com/zquestz/s/autocomplete/s-completion.bash`

Comment (zquestz) [inline README.md]: Good call, lets change it to point to the expected location.

Comment (KeizerDev) [inline README.md]: Are you gonna do this?

Comment (zquestz) [inline README.md]: Thought you had it under control. I can do later, but at work right now. =)

Comment (zquestz) [inline README.md]: Also make sure to update the comment on the completion file itself!

**Comment by KeizerDev** [comment]
No, sure! You want me to make a new PR just for that?

File: README.md (hunk 3 above)
Comment (KeizerDev) [inline README.md]: No, sure! You want me to make a new PR just for that?

Comment (zquestz) [inline README.md]: Yep go for it.

Comment (zquestz) [inline README.md]: In your PR add this to the completion file

```
if [[ -n ${ZSH_VERSION-} ]]; then
  autoload -U +X bashcompinit && bashcompinit
fi
```

This adds full zsh support, tested and working. We should update the docs and comments to reflect this.

**Comment by p-v** [comment]
Guess I missed a lot of stuff :smile: 
Great feedback.  :+1:

**Pull request merged**